daemons/gptp/linux/src/linux_hal_common.cpp

```diff
@@ -213,6 +213,9 @@ void *LinuxTimerQueueHandler( void *arg ) {
 		    struct LinuxTimerQueueActionArg *arg = iter->second;
 			timerq->timerQueueMap.erase(iter);
 			timerq->LinuxTimerQueueAction( arg );
+			if( arg->rm ) {
+				delete arg->inner_arg;
+			}
 			timer_delete(arg->timer_handle);
 			delete arg;
```

Comment:
I'm confused. Doesn't delete arg; take care of deleting arg->inner_arg ?
Also, if the code is actually correct, I think it needs a comment as to why a test for arg->rm causes a delete of arg->inner_arg.